openpype/plugins/publish/integrate_new.py

```diff
@@ -1,38 +1,88 @@
 import pyblish.api
 from avalon import io
 import openpype.api
 from datetime import datetime
-# from pype.modules import ModulesManager
 from openpype.lib.profiles_filtering import filter_profiles
-from openpype.lib import (
-    prepare_template_data,
-    create_hard_link,
-    StringTemplate,
-    TemplateUnsolved
-)
-
-# this is needed until speedcopy for linux is fixed
-if sys.platform == "win32":
-    from speedcopy import copyfile
-else:
-    from shutil import copyfile
+from openpype.lib.file_transaction import FileTransaction
 
 log = logging.getLogger(__name__)
 
 
+def assemble(files):
+    """Convenience `clique.assemble` wrapper for files of a single collection.
+
+    Unlike `clique.assemble` this wrapper does not allow more than a single
+    Collection nor any remainder files. Errors will be raised when not only
+    a single collection is assembled.
+
+    Returns:
+        clique.Collection: A single sequence Collection
+
+    Raises:
+        ValueError: Error is raised when files do not result in a single
+                    collected Collection.
+
+    """
+    # todo: move this to lib?
+    # Get the sequence as a collection. The files must be of a single
+    # sequence and have no remainder outside of the collections.
+    patterns = [clique.PATTERNS["frames"]]
+    collections, remainder = clique.assemble(files,
+                                             minimum_items=1,
+                                             patterns=patterns)
+    if not collections:
+        raise ValueError("No collections found in files: "
+                         "{}".format(files))
+    if remainder:
+        raise ValueError("Files found not detected as part"
+                         " of a sequence: {}".format(remainder))
+    if len(collections) > 1:
+        raise ValueError("Files in sequence are not part of a"
+                         " single sequence collection: "
+                         "{}".format(collections))
+    return collections[0]
+
+
+def get_instance_families(instance):
+    """Get all families of the instance"""
+    # todo: move this to lib?
+    family = instance.data.get("family")
+    families = []
+    if family:
+        families.append(family)
+
+    for _family in (instance.data.get("families") or []):
+        if _family not in families:
+            families.append(_family)
+
+    return families
+
+
+def get_frame_padded(frame, padding):
+    """Return frame number as string with `padding` amount of padded zeros"""
+    return "{frame:0{padding}d}".format(padding=padding, frame=frame)
+
+
+def get_first_frame_padded(collection):
+    """Return first frame as padded number from `clique.Collection`"""
+    start_frame = next(iter(collection.indexes))
+    return get_frame_padded(start_frame, padding=collection.padding)
+
+
+def bulk_write(writes):
+    """Convenience function to bulk write into active project database"""
+    return io._database[io.Session["AVALON_PROJECT"]].bulk_write(writes)
```

Comment:
This `bulk_write` should actually be implemented in `avalon.io` (or `openpype.io` when it's moved). Yes?